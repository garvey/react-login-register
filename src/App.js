import React from 'react';
import './App.scss';
import { Route, BrowserRouter as Router } from 'react-router-dom';
import Login from './components/auth/Login';
import Forgot from './components/auth/Forgot';
import Register from './components/auth/Register';

function App() {
  return (
    <div className="App">
      <Router>
        <Route exact path="/" component={Login} />
        <Route path="/login" component={Login} />
        <Route path="/forgot" component={Forgot} />
        <Route path="/register" component={Register} />
      </Router>
    </div>
  );
}

export default App;
