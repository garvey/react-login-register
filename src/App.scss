@import './scss/Colors.scss';
@import './components/auth/scss/Login.scss';
@import './components/auth/scss/Register.scss';
@import './scss/Layout.scss';
@import './scss/Typography.scss';
