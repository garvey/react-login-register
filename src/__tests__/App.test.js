import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import App from '../App';

it('Should render Component', () => {
  const wrapper = mount(<App />);
  expect(wrapper.debug()).toMatchSnapshot();
});
