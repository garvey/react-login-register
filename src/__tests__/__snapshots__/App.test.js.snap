// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Component 1`] = `
"<App>
  <div className=\\"App\\">
    <BrowserRouter>
      <Router history={{...}}>
        <Route exact={true} path=\\"/\\" component={[Function: Login]}>
          <Login history={{...}} location={{...}} match={{...}} staticContext={[undefined]}>
            <Container tag=\\"div\\">
              <div className=\\"container\\">
                <Row tag=\\"div\\">
                  <div className=\\"row\\">
                    <Col xs={12} md={{...}} lg={{...}} className=\\"my-auto mx-auto\\" tag=\\"div\\" widths={{...}}>
                      <div className=\\"my-auto mx-auto col-12 col-md-6 offset-md-3 col-lg-4 offset-lg-4\\">
                        <Form onSubmit={[Function]} tag=\\"form\\">
                          <form onSubmit={[Function]} className=\\"\\">
                            <div className=\\"text-center pt-4 pb-4\\">
                              <Link to=\\"/\\">
                                <LinkAnchor href=\\"/\\" navigate={[Function: navigate]}>
                                  <a href=\\"/\\" onClick={[Function: onClick]}>
                                    <img src=\\"logo.png\\" alt=\\"Company Logo\\" className=\\"img-fluid logo\\" />
                                  </a>
                                </LinkAnchor>
                              </Link>
                            </div>
                            <Card className=\\"login\\" tag=\\"div\\">
                              <div className=\\"login card\\">
                                <CardBody tag=\\"div\\">
                                  <div className=\\"card-body\\">
                                    <FormGroup tag=\\"div\\">
                                      <div className=\\"form-group\\">
                                        <Label tag=\\"label\\" widths={{...}}>
                                          <label htmlFor={[undefined]} className=\\"\\">
                                            Email
                                          </label>
                                        </Label>
                                        <input name=\\"email\\" />
                                      </div>
                                    </FormGroup>
                                    <FormGroup tag=\\"div\\">
                                      <div className=\\"form-group\\">
                                        <Label tag=\\"label\\" widths={{...}}>
                                          <label htmlFor={[undefined]} className=\\"\\">
                                            Password
                                          </label>
                                        </Label>
                                        <input name=\\"password\\" type=\\"password\\" />
                                      </div>
                                    </FormGroup>
                                    <Button outline={true} color=\\"success\\" block={true} type=\\"submit\\" tag=\\"button\\">
                                      <button type=\\"submit\\" className=\\"btn btn-outline-success btn-block\\" onClick={[Function: bound onClick]} aria-label={{...}}>
                                        Sign in
                                      </button>
                                    </Button>
                                    <FormGroup tag=\\"div\\">
                                      <div className=\\"form-group\\">
                                        <p className=\\"pt-2 text-center\\">
                                          <Link to=\\"/forgot\\">
                                            <LinkAnchor href=\\"/forgot\\" navigate={[Function: navigate]}>
                                              <a href=\\"/forgot\\" onClick={[Function: onClick]}>
                                                Forgot your password?
                                              </a>
                                            </LinkAnchor>
                                          </Link>
                                        </p>
                                      </div>
                                    </FormGroup>
                                    <FormGroup tag=\\"div\\">
                                      <div className=\\"form-group\\">
                                        <p className=\\"pt-2\\">
                                          Don&#39;t have an account?
                                           
                                          <Link to=\\"/register\\">
                                            <LinkAnchor href=\\"/register\\" navigate={[Function: navigate]}>
                                              <a href=\\"/register\\" onClick={[Function: onClick]}>
                                                Register here
                                              </a>
                                            </LinkAnchor>
                                          </Link>
                                        </p>
                                      </div>
                                    </FormGroup>
                                  </div>
                                </CardBody>
                              </div>
                            </Card>
                          </form>
                        </Form>
                      </div>
                    </Col>
                  </div>
                </Row>
              </div>
            </Container>
          </Login>
        </Route>
        <Route path=\\"/login\\" component={[Function: Login]} />
        <Route path=\\"/forgot\\" component={[Function: Forgot]} />
        <Route path=\\"/register\\" component={[Function: Register]} />
      </Router>
    </BrowserRouter>
  </div>
</App>"
`;
